src/main/java/graphql/execution/incremental/DeferredExecutionSupport.java

```diff
@@ -59,16 +65,19 @@ class DeferredExecutionSupportImpl implements DeferredExecutionSupport {
         private final ExecutionStrategyParameters parameters;
         private final ExecutionContext executionContext;
         private final BiFunction<ExecutionContext, ExecutionStrategyParameters, CompletableFuture<FieldValueInfo>> resolveFieldWithInfoFn;
+        private final BiFunction<ExecutionContext, ExecutionStrategyParameters, Supplier<ExecutionStepInfo>> executionStepInfoFn;
         private final Map<String, Supplier<CompletableFuture<DeferredFragmentCall.FieldWithExecutionResult>>> dfCache = new HashMap<>();
 
         public DeferredExecutionSupportImpl(
                 MergedSelectionSet mergedSelectionSet,
                 ExecutionStrategyParameters parameters,
                 ExecutionContext executionContext,
-                BiFunction<ExecutionContext, ExecutionStrategyParameters, CompletableFuture<FieldValueInfo>> resolveFieldWithInfoFn
+                BiFunction<ExecutionContext, ExecutionStrategyParameters, CompletableFuture<FieldValueInfo>> resolveFieldWithInfoFn,
+                BiFunction<ExecutionContext, ExecutionStrategyParameters, Supplier<ExecutionStepInfo>> executionStepInfoFn
```

Comment:
I need to be able to build the ExecutionStepInfo later and this allows me to do that back in the execution strategy